### DISCUSSION TITLE
twitter-image in app router not working but opengraph-image is

### DISCUSSION THREAD

**Discussion by jphyqr** [vercel/next.js#51305] [Help] [answered]
### Summary

created opengraph-image.tsx and twitter-image.tsx files in my app/episodes/[epsidesSlug] folder.

when sharing to facebook, this works fine and shows the dynamic image

when I share to twitter, no image/card pops up.

is there any other wiring up step you need to do for twitter besides creating this twitter-image.tsx file?

`import { ImageResponse } from 'next/server';

export const runtime = 'edge';

export const alt = 'About Acme';
export const size = {
  width: 1200,
  height: 630,
};
export const contentType = 'image/png';

export default async function Image() {
  return new ImageResponse(
    (
      <div
        style={{
          fontSize: 48,
          background: 'white',
          width: '100%',
          height: '100%',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        {`Episodes Staticall Generated`}
      </div>
    ),
    {
      ...size,
    },
  );
}
`

### Example

https://q5qb.vercel.app/episodes/david-baktiari-bussin-with-boys

**Comment by jphyqr** [comment, marked as answer]
Okay fixed this..

few things not obvious from docs

-add robots.txt to public folder so twitter can scrape your page.. this wasn't the main blocker but could be as i fixed something else

even though you have the twitter-image.txt  file it is missing a bunch of meta data needed to render.  so I added to my page.tsx file 

export const metadata: Metadata = {
  openGraph: {
    type: 'website',
  },
  twitter: {
    card: 'summary_large_image',
    title: 'Next.js',
    description: 'The React Framework for the Web',
    siteId: '1467726470533754880',
    creator: '@nextjs',
    creatorId: '1467726470533754880',
  },
};

i added the twitter{} object first and it didn't do the trick... it was the openGraph:{type:'website'} that put  me over the finish line

**Comment by techUdayMungalpara** [comment]
i also  face same proble after asking on twitter i get make both image simalar for twitter card and  opengraph it work out .
